### PR TITLE
Add KLV checksum functions

### PR DESCRIPTION
**Pull request by daniel-riehm**
The MISB KLV standards use a variety of checksum methods. This branch adds implementations of two of them ( Running 16-Bit Sum, used by [ST0601](https://gwg.nga.mil/misb/docs/standards/ST0601.17.pdf), and CRC-16-CCITT, used by [ST1108](https://gwg.nga.mil/misb/docs/standards/ST1108.3.pdf) ), accompanied by unit tests.

### CODE REVIEW

**Comment by daniel-riehm** [comment]
jenkins test this please

File: arrows/klv/klv_checksum.cxx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+/// \file
+/// Implementation of the KLV checksum functions.
+
+#include "klv_checksum.h"
+
+#include <numeric>
+#include <vector>
+
+#include <cstdlib>
```
Comment (mwoehlke-kitware) [inline arrows/klv/klv_checksum.cxx:13]: BTW, what needs `cstdlib`? If it's for `size_t`, would `cstdint` + `cstddef` be better?

File: arrows/klv/klv_checksum.cxx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+// ----------------------------------------------------------------------------
+template < class Iterator >
+uint16_t
+klv_crc_16_ccitt( Iterator data_begin, Iterator data_end )
+{
+  constexpr uint16_t initializer = 0xFFFF;
+
+  // Based on http://srecord.sourceforge.net/crc16-ccitt.html
+  auto accumulator =
+    [](uint16_t crc, uint8_t byte) -> uint16_t {
```
Comment (mwoehlke-kitware) [inline arrows/klv/klv_checksum.cxx:48]:
```suggestion
    []( uint16_t crc, uint8_t byte ) -> uint16_t {
```

File: arrows/klv/klv_checksum.cxx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+// Instantiate templates for common iterators
+#define KLV_INSTANTIATE( FN, T ) \
+  template KWIVER_ALGO_KLV_EXPORT FN< T >( T, T )
+#define KLV_INSTANTIATE_ALL( FN ) \
+  KLV_INSTANTIATE( FN, uint8_t const* ); \
+  KLV_INSTANTIATE( FN, typename std::vector< uint8_t >::iterator ); \
+  KLV_INSTANTIATE( FN, typename std::vector< uint8_t >::const_iterator )
+
+KLV_INSTANTIATE_ALL( uint16_t klv_crc_16_ccitt );
+KLV_INSTANTIATE_ALL( uint16_t klv_running_sum_16 );
```
Comment (mwoehlke-kitware) [inline arrows/klv/klv_checksum.cxx:81]: This makes me wonder if some of the recent `.txx`s could/should take this approach? (Not immediately relevant to *this* PR, though.)

**Review by mwoehlke-kitware — APPROVED** [review]
LGTM aside from the style error and optional suggestion re: use of `cstdlib`.

File: arrows/klv/klv_checksum.cxx (hunk 3 above)
Comment (daniel-riehm) [reply arrows/klv/klv_checksum.cxx:81]: My first thought was that the combinatorics of additional template parameters (e.g. integral type) would make the code rather bulky, but I guess a series of macros patterned like the ones here could be written to turn `O(xy)` into `O(x+y)`...

**Pull request merged**